llvm/include/llvm/CodeGen/MachineFrameInfo.h

```diff
@@ -72,6 +76,9 @@ class CalleeSavedInfo {
   bool isRestored()                        const { return Restored; }
   void setRestored(bool R)                       { Restored = R; }
   bool isSpilledToReg()                    const { return SpilledToReg; }
+
+  bool isHandledByTarget() const { return IsHandledByTarget; }
+  void setHandledByTarget() { IsHandledByTarget = true; }
```

Comment:
Can you just override spillCalleeSavedRegisters/restoreCalleeSavesRegisters? Why do we need a parallel customization mechanism in the same code?

Reply:
I did override those, but I'm only using them for the block VGPRs. I didn't want to copy-paste all the generic code into our target-specific methods.